projects/aca-content/src/lib/components/header-actions/header-actions.component.ts

```diff
@@ -0,0 +1,63 @@
+/*!
+ * @license
+ * Alfresco Example Content Application
+ *
+ * Copyright (C) 2005 - 2020 Alfresco Software Limited
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { ContentManagementService } from '../../services/content-management.service';
+import { PageComponent } from '../page.component';
+import { AppExtensionService } from '@alfresco/aca-shared';
+import { SetCurrentFolderAction, AppStore } from '@alfresco/aca-shared/store';
+
+@Component({
+  selector: 'aca-header-actions',
+  templateUrl: './header-actions.component.html'
+})
+export class HeaderActionsComponent extends PageComponent implements OnInit, OnDestroy {
```

Comment:
is there any reason why you inherit this component from the Page Component?

Reply:
This component is inheriting Page Component as the template of this component has buttons/actions coming from the Page Component.

This logic was moved to Page Component (Base Class) as per previous review comments